### ISSUE TITLE
v0.11 + bundles fail with API relation changed hook

### ISSUE THREAD

**Issue by lazypower** [whitmo/bundle-kubernetes#41]
These are easily fixed up with a juju resolved -r, however the response coming from kubernetes-master is that the minion is already registered.

**Comment by mbruzek** [comment]
@chuckbutler 
This could be related to a different response code. We need to look at the log messages on why the relation failed.  I pushed a fix for the different HTTP response code, to the kubernetes charms that is not vendored or maybe you didn't pull the latest kubernetes charm code from github.  What I found was the kubernetes hook would "fail" after the registration code, but were successfully registered after all.

I also noticed that the Kubernetes API itself has changed in more modern versions (greater than v0.8.1) of the api-server.  Created an issue against the kubernetes charm about this. They may be making making non backward compatible changes to the registration API in this pre v1.0 release that we need to know about and fix in the charm code.

**Comment by lazypower** [comment]
Right, and this is related to my findings @mbruzek  - the bug here is to track that with my mass bundle reorgnization that i've been working on + testing.

I've got an aggregate of the logs to dive through and query, but a grep shows it is indeed related to the status code bug. I need to rebase the subordinate conversion code w/ your latest changes and validate again.
